log/file_test.go

```diff
@@ -0,0 +1,170 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2020 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package log
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+type bytesBufferWithClose struct {
+	buffer *bytes.Buffer
+}
+
+func (bb *bytesBufferWithClose) Write(p []byte) (int, error) {
+	return bb.buffer.Write(p)
+}
```

Comment:
```suggestion
type nopCloser struct {
	Writer io.Writer
}

func (nc *nopCloser) Close() error {return nil}
```
The `Write` method is not required with an embedded struct. You can check the `NopCloser` method from the stdlib https://cs.opensource.google/go/go/+/refs/tags/go1.17.5:src/io/io.go;l=616-620